src/services/order.service.ts

```
@@ -265,59 +277,106 @@ export const updateOrderStatus = async (
+        );
       } else {
-        logger.warn(`No order items found to mark as fulfilled for orderID ${ orderId }`);
+        logger.warn(
+          `No order items found to mark as fulfilled for orderID ${orderId}`
+        );
       }
     } else {
-      logger.warn(`Unhandled order status type: ${ orderStatus } for orderID ${ orderId }`);
+      logger.warn(
+        `Unhandled order status type: ${orderStatus} for orderID ${orderId}`
+      );
     }
 
     // Update the main order status
     const updatedOrder = await Order.findByIdAndUpdate(
-      orderId, 
+      orderId,
       { status: orderStatus },
       { new: true }
     ).exec();
 
+    // Send notification to buyer when seller completes the order
+    if (
+      orderStatus.toLowerCase() === OrderStatusType.Completed.toLowerCase() &&
+      updatedOrder
+    ) {
+      try {
+        // Get seller and buyer info
+        const seller = await Seller.findById(updatedOrder.seller_id);
+        const buyer = await User.findById(updatedOrder.buyer_id);
+
+        if (seller && buyer) {
+          const message = `The seller ${
+            seller.seller_id
+          } fulfilled your order on ${new Date().toLocaleString()} and it's ready for collection/delivery.`;
+
+          // Use the existing notification API
+          await addNotification(buyer.pi_uid, message);
+
+          logger.info(
+            `Notification sent to buyer ${buyer.pi_uid} for order ${orderId}`
+          );
+        } else {
+          logger.warn(
+            `Seller or buyer not found when sending notification for order ${orderId}`
+          );
+        }
+      } catch (error) {
+        logger.error(
+          ` Error sending notification for order ${orderId}: ${error}`
+        );
+      }
+    }
+
     if (!updatedOrder) {
-      logger.error(`Order not found or failed to update status for orderID ${ orderId }`);
+      logger.error(
+        `Order not found or failed to update status for orderID ${orderId}`
```

If update failed/order not found, consider returning a typed error instead of null for clear API contract.
And, make necessary changes to handle the same on caller side.